components/tidb_query/src/codec/mysql/duration.rs

```diff
@@ -678,6 +679,37 @@ impl Duration {
         }
         buf
     }
+
+    pub fn from_i64(ctx: &mut EvalContext, mut n: i64, fsp: u8) -> Result<Duration> {
+        use crate::codec::error::{Error, ERR_TRUNCATE_WRONG_VALUE};
+
+        if n > i64::from(MAX_DURATION_VALUE) || n < -i64::from(MAX_DURATION_VALUE) {
+            // FIXME: parse as `DateTime` if `n >= 10000000000`
+            ctx.handle_overflow(Error::overflow("Duration", &n.to_string()))?;
```

Comment:
NOTE: use `handle_overflow_err` instead and no need to convert `n` to `String` manually.